src/components/interview/QuestionBank.jsx

```
@@ -0,0 +1,196 @@
+  const [selectedDifficulty, setSelectedDifficulty] = useState('');
+  const [selectedCompany, setSelectedCompany] = useState('');
+  const [selectedStatus, setSelectedStatus] = useState('');
+  const [selectedFrequency, setSelectedFrequency] = useState('');
+  
+  // Pagination State
+  const [currentPage, setCurrentPage] = useState(1);
+
+  // Filter logic
+  const filteredQuestions = useMemo(() => {
+    // Reset to page 1 on filter changes
+    setCurrentPage(1);
```

ESLint

[error] Error: Calling setState from useMemo may trigger an infinite loop

Each time the memo callback is evaluated it will change state. This can cause a memoization dependency to change, running the memo function again and causing an infinite loop. Instead of setting state in useMemo(), prefer deriving the value during render. (https://react.dev/reference/react/useState).

  25 |   const filteredQuestions = useMemo(() => {
  26 |     // Reset to page 1 on filter changes
> 27 |     setCurrentPage(1);
     |     ^^^^^^^^^^^^^^ Found setState() within useMemo()
  28 |
  29 |     return QUESTION_DATASET.filter((q) => {
  30 |       // Search text match (case-insensitive)

(react-hooks/set-state-in-render)